environment.rst

```diff
@@ -0,0 +1,81 @@
+.. _software-environment:
+
+Software Environment
+====================
+
+This page describes the software environment you will need to complete your coursework, along with some recommended development tools.
+
+The CS Software Environment
+---------------------------
+
+The Department of Computer Science has an officially supported Linux software environment that can be found on the CS Linux login servers and on the Linux machines in the CSIL labs in Crerar. All of these machines run Ubuntu 16.04 and have most of the tools and software libraries required to complete your CS coursework. Unless told otherwise by your instructor, any code you write must compile and run correctly on this software environment.
+
+However, that doesn't necessarily mean that you have to *write* your code on a CS machine (although you're welcome to do so), nor does it mean yo have to be physically sitting in front of a CS machine to compile and run your code. There are multiple mechanisms to write your code on your own computer (using other software environments, including Windows and Mac operating systems) and to access a CS software environment remotely, all of which are discussed in the :ref:`working-remotely` page.
```

Comment:
typo (missing 'u'): "nor does it mean yoU have to be physically..."

Reply:
Fixed.